doc/source/reference/c-api.types-and-structures.rst

```diff
@@ -355,13 +355,14 @@ PyArrayDescr_Type
 
 .. c:member:: PyObject *PyArray_Descr.fields
 
-    If this is non-NULL, then this data-type-descriptor has fields
+    For built-in data types, this is set to ``Py_None``. Otherwise,
```

Comment:
Agree about the wording. I was trying to figure out what all the permutations were ;)